include/seastar/net/socket_defs.hh

```diff
@@ -53,13 +58,17 @@ public:
     socket_address(ipv4_addr);
     socket_address(const ipv6_addr&);
     socket_address(const net::inet_address&, uint16_t p = 0);
+    socket_address(const std::filesystem::path);
```

Comment:
it's nice to support `std::filesystem::path`. but seastar support C++14. so might want to use `seastar::compat::filesystem::path` ? or how about just using `std::sstring` ? as we are not using any specific feature of `std::filesystem::path`, right?

Reply:
Indeed I am not using any feature. Will remove the fs::path.
Note that scylladb::seastar does use <filesystem> or <experimental/filesystem>

Reply:
that's why i suggest use `seastar::compat::filesystem::path` as an option.